Automattic/wp-calypso
# Plans Next: Untangle Features Grid from media queries, up reuse and default responsiveness

Related to https://github.com/Automattic/wp-calypso/issues/79010

## Proposed Changes

This is part of a series of PRs for untangling the pricing grids/components from screen/media queries and any Calypso-dependent CSS context (such as sidebar width). The end goal is to (1) make the components responsive and reusable in any context, and (2) to be able to render the grids anywhere on a page (in a modal, sidebar, a half-page column, etc.). So pdgrnI-2uQ-p2 becomes relevant and this work should set the foundations for it.

In this PR:
- we untangle the Features Grid from media queries
- we do not rewrite the Features Grid implementation (it's an old-fashioned table, and there's a separate component hierarchy for mobile views - all the stuff we "flagged" in https://github.com/Automattic/wp-calypso/issues/72910), but rather hack our way through the existing setup
- the approach is akin to "container queries", which we may consider surfacing down the line (assuming support is not an issue)
- also introduces `isInAdmin` property - on several occasions, this has been the exception we should have been conditioning on, and not `isInSignup` (since logged-out will behave similarly to signup)

This also fixes `/jetpack-app/plans`, which looks awfully broken now on anything other than mobile (but that might be the intention I guess)

## Testing Instructions



* Try several flows along with `/plans[ site ]` and main `/start/plans`
* Confirm the grid resizes properly and gives the various views (mobile, tablet, desktop)
    * The above may not be as precise as previously, but they should be rendering the grid without any overflows of text, like the plan titles/names. So if the grid fits, then we should be good
* Confirm margins and the rest of the layout are correct in `/setup` and `/start` across the various views (mobile, tablet, desktop)
* Confirm text sizes are correct across the various views (mobile, tablet, desktop)

## Pre-merge Checklist
